packages/plugins/jwt/src/index.ts

```diff
@@ -0,0 +1,166 @@
+import { JwksClient } from 'jwks-rsa'
+import jsonwebtoken, { JwtPayload, Algorithm } from 'jsonwebtoken'
+import { Plugin, createGraphQLError } from 'graphql-yoga'
+
+const { decode } = jsonwebtoken
+
+export interface JwtPluginOptions {
+  /**
+   * The endpoint to fetch keys from.
+   *
+   * For example: https://example.com/.well-known/jwks.json
+   */
+  jwksUri?: string
+  /**
+   * Signing key to be used to verify the token
+   * You can also use the jwks option to fetch the key from a JWKS endpoint
+   */
+  signingKey?: string
+  /**
+   * List of the algorithms used to verify the token
+   *
+   * Default: ["HS256"] (Allowed values: HS256, HS384, HS512, RS256, RS384, RS512, ES256, ES384, ES512, PS256, PS384, PS512, none)
+   */
+  algorithms?: Algorithm[]
+  /**
+   * The audience is the identifier of the API and is forwarded to your Auth API in order to specify for which API we are trying to authenticate our user for. E.g. if our API is hosted on `http://localhost:3000/graphql`, we would pass that value.
+   */
+  audience?: string
+  /**
+   * For example: `https://myapp.auth0.com/`
+   */
+  issuer: string
+  /**
+   * Once a user got successfully authenticated the authentication information is added on the context object under this field. In our resolvers we can then access the authentication information via `context._jwt.sub`.
+   */
+  extendContextField?: string
+  /**
+   * Function to extract the token from the request object
+   *
+   * Default: Extracts the token from the Authorization header with the format `Bearer <token>`
+   */
+  getToken?: (params: {
+    request: Request
+    serverContext: object | undefined
+    url: URL
+  }) => string | undefined
+}
+
+export function useJwt(options: JwtPluginOptions): Plugin {
+  if (!options.signingKey && !options.jwksUri) {
+    throw new TypeError('You need to provide either a signingKey or a jwksUri')
+  }
+
+  if (options.signingKey && options.jwksUri) {
+    throw new TypeError(
+      'You need to provide either a signingKey or a jwksUri, not both',
+    )
+  }
```

Comment:
We can apply this constraint via TypeScript types instead of making it runtime errors

Reply:
Good point, this should be reflected in types. But anyway, since we have JS users, we probably want to check for it at runtime ?